lib/string.c

```diff
@@ -53,7 +53,7 @@ void *memmove(void *dest, const void *src, size_t n)
     return dest;
 }
 
-#ifdef DEBUG_GDB
+#if (defined(DEBUG_GDB) || defined(__loongarch_lp64))
```

Comment:
Having to do this looks like a compiler deficiency, but then these don't work on Clang either ( #405 ).

Reply:
Yes, I think it would be better to implement the string family functions ourselves? If so, we can give up the reliance on compiler builtin functions.

Reply:
In x86 and x86-64 release builds, GCC's builtins usually produce good, tiny code: `rep stos[bwlq]` for `memset`, `rep movs[bwlq]` for `memcpy`. A while ago, when I tried to use these builtins to replace calls to generic, out of line functions, I'm pretty sure that the code was also better than inline versions of the current out of line implementations, and I didn't feel like adding complexity to the latter in order to yield better code, so I simply switched memtest86+ to the GCC builtins. There are now 3 (known) situations where the builtins can't be used, though... meh.

Reply:
So, I suggest, this two functions can also be used by some ARCH who not complete it in GCC built-in functions. But Clang... :) .